src/components/PopupWithForm.js

```diff
@@ -6,6 +6,7 @@ class PopupWithForm extends Popup {
     this._handleFormSubmit = handleFormSubmit;
     this._form = document.querySelector(popupSelector).querySelector("form");
```

Comment:
`document.querySelector(popupSelector)`

**No** need to find the popup again in the inheritors of class `Popup` because the parent has already found it in its constructor and it’s `this._popupElement`. It’s available in any inheritor.


```js

    this._form = this._popupElement.querySelector("form");

```